packages/cli/src/scaling/job-processor.ts

```diff
@@ -0,0 +1,182 @@
+import { Service } from 'typedi';
+import { BINARY_ENCODING, ApplicationError, Workflow } from 'n8n-workflow';
+import { WorkflowExecute } from 'n8n-core';
+import { Logger } from '@/Logger';
+import config from '@/config';
+import { ExecutionRepository } from '@/databases/repositories/execution.repository';
+import { WorkflowRepository } from '@/databases/repositories/workflow.repository';
+import * as WorkflowExecuteAdditionalData from '@/WorkflowExecuteAdditionalData';
+import { NodeTypes } from '@/NodeTypes';
+import type { ExecutionStatus, IExecuteResponsePromiseData, IRun } from 'n8n-workflow';
+import type { Job, JobId, JobResult, RunningJob, RunningJobSummary } from './types';
+import type PCancelable from 'p-cancelable';
+
+/**
+ * Responsible for processing jobs from the queue, i.e. running enqueued executions.
+ */
+@Service()
+export class JobProcessor {
+	private readonly runningJobs: { [jobId: JobId]: RunningJob } = {};
```

Comment:
This would be more readable:

```suggestion
	private readonly runningJobs: Record<JobId, RunningJob> = {};
```

Reply:
Sorry missed this one, will address in next round 👍🏻